src/utils/find.ts

```diff
@@ -0,0 +1,12 @@
+import { Account, EnvRegExp } from '../project';
+
+export function findAccountTypeByEnvironment(obj: Record<string, Account>, predicate: string): string | undefined {
+  for (const key in obj) {
+    if (obj.hasOwnProperty(key)) {
+      if (obj[key].environments?.some((environment) => new EnvRegExp(environment).test(predicate))) {
+        return key;
+      }
+    }
+  }
```

Comment:
Just a matter of preference: I would probably use `Object.entries` for this myself:
```ts
Object
  .entries(obj)
  .filter(([key, value]) => value.environments?.some((environment) => new RegExp(environment).test(predicate)))
  .map(([key]) => key).at(0);
```

The benefit of this is that there's no need to use `hasOwnProperty` check as `Object.entries` only returns object's own properties.

But again, this is just more about _personal taste_ than actual "code quality".

Reply:
I have changed my preference. I like the less`for` approach.